client/components/LearnerViewDirective.js

```diff
@@ -771,8 +804,14 @@ tie.directive('learnerView', [function() {
             var feedbackParagraphs = feedback.getParagraphs();
             var hasSyntaxError = false;
             for (var i = 0; i < feedbackParagraphs.length; i++) {
+              clearHighlight();
               if (feedbackParagraphs[i].isSyntaxErrorParagraph()) {
                 hasSyntaxError = true;
+                var errorContentArray = feedbackParagraphs[i].getContent()
+                  .split(' ');
+                var syntaxErrorLineNumber = errorContentArray[errorContentArray
```

Comment:
Also I think this is better done by calling something like feedbackParagraphs[i].getErrorLineNumber() -- it's something that the paragraph object itself should take care of.